frontend/appflowy_flutter/lib/mobile/presentation/database/board/mobile_board_page.dart

```diff
@@ -58,50 +102,63 @@ class _MobileBoardContentState extends State<MobileBoardContent> {
     );
 
     return BlocListener<BoardBloc, BoardState>(
-      listenWhen: (previous, current) =>
-          previous.recentAddedRowMeta != current.recentAddedRowMeta,
+      listenWhen: (previous, current) => current.isOpenCard,
       listener: (context, state) {
-        context.push(
-          MobileRowDetailPage.routeName,
-          extra: {
-            MobileRowDetailPage.argRowId: state.recentAddedRowMeta!.id,
-            MobileRowDetailPage.argDatabaseController:
-                context.read<BoardBloc>().databaseController,
+        state.maybeWhen(
+          orElse: () {},
+          openCard: (rowMeta) {
+            context.push(
+              MobileRowDetailPage.routeName,
+              extra: {
+                MobileRowDetailPage.argRowId: rowMeta.id,
+                MobileRowDetailPage.argDatabaseController:
+                    context.read<BoardBloc>().databaseController,
+              },
+            );
           },
         );
       },
       child: BlocBuilder<BoardBloc, BoardState>(
         builder: (context, state) {
-          final showCreateGroupButton =
-              context.read<BoardBloc>().groupingFieldType.canCreateNewGroup;
-          final showHiddenGroups = state.hiddenGroups.isNotEmpty;
-          return AppFlowyBoard(
-            boardScrollController: scrollManager,
-            scrollController: scrollController,
-            controller: context.read<BoardBloc>().boardController,
-            groupConstraints: BoxConstraints.tightFor(width: screenWidth * 0.7),
-            config: config,
-            leading: showHiddenGroups
-                ? MobileHiddenGroupsColumn(
-                    padding: config.groupHeaderPadding,
-                  )
-                : const HSpace(16),
-            trailing: showCreateGroupButton
-                ? const MobileBoardTrailing()
-                : const HSpace(16),
-            headerBuilder: (_, groupData) => BlocProvider<BoardBloc>.value(
-              value: context.read<BoardBloc>(),
-              child: GroupCardHeader(
-                groupData: groupData,
-              ),
-            ),
-            footerBuilder: _buildFooter,
-            cardBuilder: (_, column, columnItem) => _buildCard(
-              context: context,
-              afGroupData: column,
-              afGroupItem: columnItem,
-              cardMargin: config.cardMargin,
-            ),
+          return state.maybeMap(
+            orElse: () => const SizedBox.shrink(),
```

Comment:
Should this have an info widget?
```suggestion
            orElse: () => FlowyMobileStateContainer.info(
              emoji: '🛸',
              title: LocaleKeys.board_mobile_failedToLoad.tr(),
            ),
```

Reply:
Correct me if I'm wrong but as I understand it this case isn't reachable? The parent widget should handle the other cases